lib/CryptoPluginApi/extensionplugins.cpp

```diff
@@ -15,10 +15,642 @@
+
+  Plugin implementers must be aware that the information reporting methods
+  (encryptionType(), and canStoreKeys()) will be invoked from the main
+  thread of the secrets daemon, while the various interface operation methods
+  will be invoked from a separate thread.  Plugins are loaded and plugin
+  instances are constructed in the main thread.
+
+  In order to implement a Crypto extension plugin, plugin implementers should
+  specify the following in their .pro file:
+  \code
+  CONFIG += link_pkgconfig
+  PKGCONFIG += sailfishcryptopluginapi
+  \endcode
+
+  The CryptoPlugin class extends the Sailfish::Secrets::PluginBase abstract
+  base class, and the sailfishcryptopluginapi pkgconfig file will entail
+  a dependency upon sailfishsecretspluginapi and ailfishsecrets, as
+  well as sailfishcrypto.
+ */
+
+/*!
+ * \enum CryptoPlugin::EncryptionType
+ *
+ * This enum defines the types of encryption capability which may be offered by plugins
+ *
+ * \value NoEncryption No encryption is performed
+ * \value SoftwareEncryption Encryption is performed by "normal" rich execution environment application
+ * \value TrustedExecutionSoftwareEncryption Encryption is performed by trusted execution environment application
+ * \value SecurePeripheralEncryption Encryption is performed by a secure hardware peripheral via TEE application
+ */
+
+/*!
+ * \brief Construct a new CryptoPlugin instance
+ */
 CryptoPlugin::CryptoPlugin()
 {
 }
 
+/*!
+ * \brief Clean up any memory associated with the CryptoPlugin instance
+ */
 CryptoPlugin::~CryptoPlugin()
 {
 }
+
+/*!
+ * \fn CryptoPlugin::canStoreKeys() const
+ * \brief Returns true if the plugin can store keys
+ *
+ * If the plugin either exposes some built-in keys (e.g. as in the case of
+ * a USB-token-backed plugin) or allows the creation and storage of new keys
+ * (e.g. as in the case of an online-service-backed plugin), it should
+ * return true from this method and also must implement the
+ * Sailfish::Secrets::EncryptedStoragePlugin interface.
+ */
+
+/*!
+ * \fn CryptoPlugin::encryptionType() const
+ * \brief Returns the type of encryption capability offered by the plugin
+ */
+
+/*!
+ * \fn CryptoPlugin::generateRandomData(quint64 callerIdent, const QString &csprngEngineName, quint64 numberBytes, const QVariantMap &customParameters, QByteArray *randomData)
+ * \brief Writes \a numberBytes of random data generated using the random
+ *        number generator identified by the specified \a csprngEngineName
+ *        into \a randomData for the caller identified by the given
+ *        \a callerIdent.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Crypto::Result with the result code set to
+ * Sailfish::Crypto::Result::Failed and the error code set to
+ * Sailfish::Crypto::Result::CryptoPluginIsLockedError.
+ *
+ * The \a customParameters will contain plugin-specific parameters which may
+ * be required by the plugin.  Such parameters must be documented for clients
```

Comment:
Two spaces.

Reply:
I tend to manually typeset with double space to emulate m-dash space equivalence after periods.  qdoc ignores such whitespace only, so it's only an issue for string literals or translation strings.